src/features/deployProcess/components/Stepper/genericStepper.tsx

```
@@ -1,141 +1,219 @@
+    }
 
     const serverStepIndex = stepsConfig.findIndex(
-      (s) => s.label === product?.CreationStatus
+      (s) => s.title === product.CreationStatus
     );
 
-    if (
-      productId &&
-      product &&
-      currentStepIndex > serverStepIndex
-    ) {
+    if (currentStepIndex > serverStepIndex) {
+      console.log('[Stepper] Advancing product step on server', {
+        productId,
+        from: serverStepIndex,
+        to: currentStepIndex,
+      });
+      // Update server with new step index
       dispatch(updateProductStep({
         productId,
         stepNumber: currentStepIndex + 1
       }));
+    } else {
     }
   }, [currentStepIndex, dispatch, product]);
 
-  const handleCompleteStep = useCallback(() => {
-    const userProductKey = getUserProductKey();
-    const productId = localStorage.getItem(userProductKey);
-    if (!productId) return;
 
-    dispatch(markStepCompleted(currentStepIndex));
-    dispatch(updateProductStep({ productId, stepNumber: currentStepIndex + 1 }));
+  const [canGoNext, setCanGoNext] = useState(false);
 
-    if (currentStepIndex === stepsConfig.length - 1) {
-      setShowFinalPopup(true);
+  // Reset canGoNext on every new step
+  useEffect(() => {
+    setCanGoNext(false);
+  }, [currentStepIndex]);
+
+  // Render the current step component based on currentStepIndex  
+  const CurrentStepComponent = currentStepIndex !== null ? stepsConfig[currentStepIndex]?.component : null;
+
+  const handleCompleteStep = useCallback(async () => {
+    if (currentStepIndex === null) {
+      return;
     }
-  }, [dispatch, currentStepIndex]);
 
-  const handleNext = () => {
-     if (completedSteps[currentStepIndex] && currentStepIndex < stepsConfig.length - 1) 
-    {
-      const nextKey = getStepKeyByIndex(currentStepIndex + 1);
+    const validateStepFn = stepsConfig[currentStepIndex]?.validateStep;
+
+    if (validateStepFn) {
+      try {
+        const valid = await validateStepFn();
+        if (!valid) {
```

Consider notifying user when validateStep fails instead of silent return